src/serverlist.cpp

```diff
@@ -848,6 +848,15 @@ void CServerListManager::OnTimerCLRegisterServerResp()
     }
 }
 
+void CServerListManager::OnAboutToQuit()
+{
+    // Sets the lock
```

Comment:
Is this comment correct here? Doesn't the lock get set below?

Reply:
It's a comment about the line that immediately follows it.  `Unregister()` sets the lock.  It's _why_ the lock isn't taken earlier.

Reply:
Ok. Wasn’t sure if Unregister() sets a lock. It’s a bit unintuitive from the name so it’s not obvious.